diff --git a/src/qmath/lqmath-104/lqmath.c b/src/qmath/lqmath-104/lqmath.c
index fd1bbef..dab4541 100644
--- a/src/qmath/lqmath-104/lqmath.c
+++ b/src/qmath/lqmath-104/lqmath.c
@@ -257,7 +257,8 @@ static int Lpow(lua_State *L)			/** pow(x,y) */
  if (b<0)
  {
   b=-b;
-  report(L,mp_rat_recip(a,a),0);
+  report(L,mp_rat_recip(a,c),0);
+  return report(L,mp_rat_expt(c,b,c),1);
  }
  return report(L,mp_rat_expt(a,b,c),1);
 }
